raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -64,129 +68,192 @@ def events_for_cancel_current_route(transfer_description) -> List[Event]:
+        state_change=state_change,
+        channelidentifiers_to_channels=channelidentifiers_to_channels,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
 
-        msg = 'The previous transfer must be cancelled prior to trying a new route'
-        assert payment_state.initiator is None, msg
 
+def handle_init(
+        payment_state: InitiatorPaymentState,
+        state_change: ActionInitInitiator,
+        channelidentifiers_to_channels: ChannelMap,
+        pseudo_random_generator: random.Random,
+        block_number: BlockNumber,
+) -> TransitionResult:
+    events: List[Event]
+    if payment_state is None:
         sub_iteration = initiator.try_new_route(
-            old_initiator_state=old_initiator_state,
             channelidentifiers_to_channels=channelidentifiers_to_channels,
             available_routes=state_change.routes,
-            transfer_description=transfer_description,
+            transfer_description=state_change.transfer,
             pseudo_random_generator=pseudo_random_generator,
             block_number=block_number,
         )
 
-        events.extend(cancel_events)
-        events.extend(sub_iteration.events)
-        assert sub_iteration.new_state
-        payment_state.initiator = sub_iteration.new_state
+        events = sub_iteration.events
+        if sub_iteration.new_state:
+            payment_state = InitiatorPaymentState(
+                initiator_transfers={
+                    sub_iteration.new_state.transfer.lock.secrethash: sub_iteration.new_state,
+                },
+            )
+    else:
+        events = list()
 
     iteration = TransitionResult(payment_state, events)
-
     return iteration
 
 
 def handle_cancelpayment(
         payment_state: InitiatorPaymentState,
-        channel_state: NettingChannelState,
+        channelidentifiers_to_channels: ChannelMap,
 ) -> TransitionResult:
     """ Cancel the payment and all related transfers. """
     # Cannot cancel a transfer after the secret is revealed
-    if can_cancel(payment_state):
-        transfer_description = payment_state.initiator.transfer_description
-        cancel_events = cancel_current_route(payment_state)
-
-        cancel = EventPaymentSentFailed(
-            payment_network_identifier=channel_state.payment_network_identifier,
-            token_network_identifier=channel_state.token_network_identifier,
-            identifier=transfer_description.payment_identifier,
-            target=transfer_description.target,
-            reason='user canceled payment',
-        )
-        cancel_events.append(cancel)
+    events = list()
+    for secrethash in list(payment_state.initiator_transfers.keys()):
+        initiator_state = payment_state.initiator_transfers[secrethash]
+        channel_identifier = initiator_state.channel_identifier
+        channel_state = channelidentifiers_to_channels.get(channel_identifier)
+
+        if not channel_state:
```

Comment:
this doesn't look right, if the channel does not exist at very least settle has been called, so the transfer task should not exist anymore.

IOW, if this happens clean up is missing somewhere else, probably the state change `ContractReceiveChannelSettled` is not handled.

Reply:
Hummm, because of the above it makes sense to enforce a lock timeout smaller than the channel settlement timeout (this is not necessary for the initiator at the protocol level, point is that it makes sense at the code level)